### ISSUE TITLE
az kusto database-principal-assignment create doesn't work with role UnrestrictedViewers

### ISSUE THREAD

**Issue by vladris** [Azure/azure-cli#16469]
### Kusto

I'm trying to use the ``az kusto database-principal-assignment`` command to grant *UnrestrictedViewers* access but I get the following error: ``(BadRequest) Database principal role is invalid. Please provide valid database principal role (Admin, Viewer, User, etc.).``.

Here is the full command for context:

```
$me = az ad signed-in-user show --query objectId

az kusto database-principal-assignment create `
--cluster-name "mycluster" `
--database-name "mydatabase"`
--principal-id $me `
--principal-type User `
--role UnrestrictedViewers `
--principal-assignment-name "assignmentname" `
--resource-group "myrg"
```

Running ``az kusto database-principal-assignment create -h`` shows UnrestrictedViewers as one of the allowed values for role, but looks like the server rejects it.

**Comment by yonzhan** [comment]
Thanks for the feedback! We are routing this to the appropriate team for follow-up. cc @radennis.

**Comment by vladris** [comment]
So any thoughts on this?

**Comment by radennis** [comment]
Adding @ilayrn and @orhasban

**Comment by astauben** [comment]
Hi @vladris it is indeed bug in our clients, the role name is not supported in the server side. 
We will fix this issue, and future clients will have the right property name. 

In the meanwhile, would you like us add the role to your database manually?

**Comment by vladris** [comment]
No, that's OK. Thanks for offering. I know how to add roles manually, was just looking at Azure CLI automation so I'll wait for the fix.

**Comment by astauben** [comment]
OK. Thanks for reporting this bug.

**Comment by vladris** [comment]
Any idea when this will be fixed?

**Comment by astauben** [comment]
We are planning to release new client (with the fix) around mid March.
I'll update the thread once the new client is released.
